tests/src/test/scala/spire/PartialSyntaxTest.scala

```diff
@@ -0,0 +1,40 @@
+package spire
+
+import spire.algebra._
+import spire.algebra.partial._
+import spire.optional.partialIterable._
+import spire.std.int._
+import spire.util._
+
+import org.scalatest.{FunSuite, Matchers}
+import org.scalatest.prop.Checkers
+
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalacheck.Arbitrary._
+import org.scalacheck.Prop._
+
+class PartialSyntaxTest extends FunSuite with Checkers with BaseSyntaxTest {
+
+  implicit val IntGroup: Group[Int] = implicitly[AdditiveGroup[Int]].additive
+  implicit val SeqIntEq: Eq[Seq[Int]] = spire.optional.genericEq.generic[Seq[Int]]
+
+  test("Semigroupoid syntax")(check(forAll { (a: Seq[Int], b: Seq[Int]) => testSemigroupoidSyntax(a, b) }))
+  test("Groupoid syntax")(check(forAll { (a: Seq[Int], b: Seq[Int]) => testGroupoidSyntax(a, b) }))
+
+  def testSemigroupoidSyntax[A: Semigroupoid: Eq](a: A, b: A) = {
+    import spire.syntax.semigroupoid._
+    ((a |+|? b) == Semigroupoid[A].partialOp(a, b)) &&
```

Comment:
Since we are using `Eq` here we should probably be using `===` instead of `==` to test equality.